### DISCUSSION TITLE
Converting to vec4 in Screen Shader not giving expected results

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#208] [Q&A] [answered]
I am repeatedly coming across this weird behavior when using Screen Shaders. Here is an axample:
![image](https://user-images.githubusercontent.com/88381426/146778938-18f50ff5-9ad2-4220-bb43-d70700f5869a.png)
As you can see, the entire screen is cyan-colored. This would mean that the red channel, that I am pluggin into the vec4_join node is black (0.0) all across the image. But if I plug the sampler output directly into the Screen Shader output you can see that that can not be possible, as seen below:
![image](https://user-images.githubusercontent.com/88381426/146779247-ec4c086d-d592-4a15-b949-3e088bffce20.png)
I would expect in the first image to see a little contour of the scene imprinted in the otherwise pretty cyan-looking image. 
The weird thing is that when I go back into the first example and decrease the 'a' input on the vec4_join node, I will start to see some of the effect I would expect:
![image](https://user-images.githubusercontent.com/88381426/146779473-b78f1d24-22dd-4461-a198-9486f04f7cb2.png)
Its subtle but you can see the scene come through a little.

**Comment by pragma37** [comment, marked as answer]
Long story short:
You should plug the alpha channel into vec4 join.

Explanation:
The Render Layer nodes are applied to both opaque and transparent layers.
(see https://github.com/bnpr/Malt/tree/Development/Malt/Pipelines/NPR_Pipeline#render-overview)
So you are basically only seeing the result of the last transparent layer. If you set Transparency > Layers to 0 in the settings you will see the result you expected.
